editoast/fga_migrations/src/migration.rs

```diff
@@ -0,0 +1,509 @@
+
+#[cfg(not(test))]
+static OPENFGA_MIGRATIONS: LazyLock<IndexMap<&'static str, &'static str>> = LazyLock::new(|| {
+    // The migrations declaration order matches the order in which they will be applied
+    indexmap! {
+        "initial_model" => include_str!("../migrations/1_initial_model.fga"),
+    }
+});
+
+#[cfg(test)]
+static OPENFGA_MIGRATIONS: LazyLock<IndexMap<&'static str, &'static str>> = LazyLock::new(|| {
+    // The migrations declaration order matches the order in which they will be applied
+    indexmap! {
+        "initial_model" => include_str!("../tests/migrations/1_initial_model.fga"),
+        "intermediate_model" => include_str!("../tests/migrations/2_intermediate_model.fga"),
+        "final_model" => include_str!("../tests/migrations/3_latest_model.fga"),
+    }
+});
+
+#[derive(Debug, Clone)]
+pub struct Settings {
+    pub migration_settings: MigrationSettings,
+    pub telemetry: TelemetrySettings,
+}
+
+#[derive(Debug, Clone)]
+pub struct MigrationSettings {
+    pub connection_settings: ConnectionSettings,
+    pub migration_store: String,
+    pub authorization_store: String,
+}
+
+#[derive(Debug, Clone)]
+pub struct TelemetrySettings {
+    pub endpoint: Url,
+}
+
+impl From<&CliParser> for Settings {
+    fn from(client: &CliParser) -> Self {
+        let connection_settings = ConnectionSettings::new(
+            client.openfga_url.host().unwrap().to_string(),
+            client.openfga_url.port_or_known_default().unwrap(),
+            fga::client::Limits {
+                max_checks_per_batch_check: client.tuple_reads,
+                max_tuples_per_write: client.tuple_writes,
+            },
+        );
+        Settings {
+            migration_settings: MigrationSettings {
+                connection_settings,
+                migration_store: client.migrations_store.clone(),
+                authorization_store: client.authorization_store.clone(),
+            },
+            telemetry: TelemetrySettings {
+                endpoint: url::Url::parse(&client.telemetry_endpoint).expect("Invalid url"),
+            },
+        }
+    }
+}
+
+#[derive(fga::Type, fga::User, derive_more::FromStr, Debug, Clone, PartialEq, Eq, Hash)]
+struct Migration(pub usize);
+
+#[derive(fga::Type, fga::Object, derive_more::FromStr, Debug, Clone, PartialEq, Eq, Hash)]
+struct Model(pub String);
+
+fga::relations! {
+    Model {
+        apply: Migration
+    }
+}
+
+#[derive(Debug, thiserror::Error)]
+pub enum MigrationError {
+    #[error("Invalid migration name `{0}`")]
+    InvalidMigrationName(String),
+    #[error(transparent)]
+    InitializationError(#[from] InitializationError),
+    #[error(transparent)]
+    QueryError(#[from] QueryError),
+    #[error(transparent)]
+    DeleteTuplesError(#[from] Either<RequestFailure, TooManyTuples>),
+    #[error(transparent)]
+    RequestFailure(#[from] RequestFailure),
+}
+
+#[instrument(name = "migration-run")]
```

Comment:
```suggestion
#[instrument(name = "migration_run")]
```

nit: we mostly use underscores (not my preference either, but that's life)